packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -90,23 +85,15 @@ export class Wallet {
 
     const {multiple, initialDelay, numberOfAttempts} = this._retryOptions;
     for (let i = 0; i < numberOfAttempts; i++) {
-      if (remaining.size === 0) return;
-
+      if (isComplete) return {type: 'Success'};
       const delayAmount = initialDelay * Math.pow(multiple, i);
       await delay(delayAmount);
 
-      const {outbox} = await this._engine.syncObjectives(objectives.map(o => o.objectiveId));
+      const {outbox} = await this._engine.syncObjectives([objective.objectiveId]);
       await this._messageService.send(getMessages(outbox));
```

Comment:
Am I right to believe that in order to remove the `EventEmitter` behaviour from the `Engine` class, that the `MessageService` would have to collect new objectives created, and return a result?

Reply:
I guess the `MessageService` could also be an event emitter?